src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3636,17 +3638,19 @@ protected override void VisitStatement(BoundStatement statement)
                         Visit(initializer.Arguments[0]);
                         break;
                     case BoundCollectionExpressionSpreadElement spread:
+                        Visit(spread);
+                        if (elementType.HasType &&
+                            spread.ElementPlaceholder is { } &&
+                            spread.IteratorBody is { })
                         {
-                            Visit(spread);
-                            if (elementType.HasType && spread.ExpressionPlaceholder is { })
-                            {
-                                var itemState = spread.EnumeratorInfoOpt == null ? default : ResultType;
-                                var completion = VisitOptionalImplicitConversion(spread.ExpressionPlaceholder, elementType,
-                                    useLegacyWarnings: false, trackMembers: false, AssignmentKind.Assignment, delayCompletionForTargetType: true,
-                                    operand => itemState).completion;
-                                Debug.Assert(completion is not null);
-                                elementConversionCompletions.Add(completion);
-                            }
+                            var itemResult = spread.EnumeratorInfoOpt == null ? default : _visitResult;
+                            var iteratorBody = ((BoundExpressionStatement)spread.IteratorBody).Expression;
+                            AddPlaceholderReplacement(spread.ElementPlaceholder, expression: null, itemResult);
```

Comment:
What's the significance of using a `null` replacement expression here, versus, say, using the same placeholder node as the replacement, which seems to be done in some other places.

Reply:
Thanks, I didn't realize we used the placeholder as the replacement in other locations. Updated.